### PR TITLE
remote python client for podman

### PR DESCRIPTION
**Pull request by jwhonce**
* Use podman library for access
* Verbose error checking
* Planned windows and macosx ports

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
how would you feel about adding a README.md to this with a brief how to use it?

**Comment by baude** [comment]
would you consider using an entrypoint in setup.py?  something like -> https://github.com/projectatomic/registries/blob/master/setup.py#L12

**Comment by baude** [comment]
couple of comments, LGTM, nice work @jwhonce

**Comment by rhatdan** [comment]
What is the command used to run this?  Could we get at least a simple man page.

pypodman would be my preferred command name?  But I am willing to look at others.
Also want to have a

make install.python 
Option to install this.

**Comment by jwhonce** [comment]
@baude @rhatdan Agree to all.  I wanted to get this in front of people early. I added WIP to mark it as such.

**Comment by rhatdan** [comment]
@jwhonce Looks good, and I would like to merge this to allow people to play with it.  But we need an easy way to install and execute the command.

**Comment by baude** [comment]
yeah what he said ... id like to test before i hit the road too, maybe in osx for you

**Comment by jwhonce** [comment]
@baude last commit has setup.py entry-point configured

**Comment by rhatdan** [comment]
@jwhonce Conflicts?
Remove the [WIP] on the PR, and lets get this in.

**Comment by jwhonce** [comment]
@rhatdan for both podman and pypodman, you now have 

make clean && make install

both work on ~/.local/lib/python...

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Comment by mheon** [comment]
I'm going to give it 5 minutes and force-merge this and #1089 so I can cut a release

File: contrib/spec/podman.spec.in [hunk 1]
```diff
@@ -389,7 +405,12 @@ BUILDTAGS=$BUILDTAGS make binaries docs
 
 %if %{with varlink}
```
Comment (lsm5) [inline contrib/spec/podman.spec.in]: export ARCH=$(uname -m)

File: contrib/spec/podman.spec.in [hunk 2]
```diff
@@ -389,7 +405,12 @@ BUILDTAGS=$BUILDTAGS make binaries docs
...
 #untar contents for python-podman
-pushd contrib/python/dist
+pushd contrib/python/podman/dist
+tar zxf %{name}*.tar.gz
```
Comment (lsm5) [inline contrib/spec/podman.spec.in]: tar zxf %[name}-%{version}-linux-$ARCH.tar.gz

File: contrib/spec/podman.spec.in [hunk 3]
```diff
@@ -389,7 +405,12 @@ BUILDTAGS=$BUILDTAGS make binaries docs
...
+popd
+
+#untar contents for python-pypodman
+pushd contrib/python/pypodman/dist
 tar zxf %{name}*.tar.gz
```
Comment (lsm5) [inline contrib/spec/podman.spec.in]: should be `tar zxf py%{name}-%{version}-linux-$ARCH.tar.gz`

Comment (lsm5) [reply contrib/spec/podman.spec.in]: we also need a line in the %files section (order doesn't really matter) perhaps right below the %{_bindir}/%{name} entry like so: `%{_bindir}/py%{name}`

**Comment by lsm5** [comment]
setup.py install --root $BUILDROOT seems to install `/usr/lib/python3.6/site-packages/lib` <- which seems wrong, it should atleast be in /usr/lib/python3.6/site-packages/podman/lib or something

**Comment by lsm5** [comment]
@jwhonce would it make sense to move `contrib/python/pypodman/lib` to say `contrib/python/pypodman/pypodman/libs` ... kinda like what's being done for `contrib/python/podman/podman/libs`. I think that'd make packaging easier.

**Pull request merged**